### PR TITLE
Removing MetricsLogger and OpenSearch analytics cluster

### PR DESCRIPTION
**Pull request by gregschohn**
### Description

Kill the OpenSearch analytics domain and its bastion host that came along with the otel-collector, leaving only the collector (which will shoot metrics and traces to AWS).
I've also updated the README and have made the cdk.context.json set otelCollectorEnabled = true, though maybe the default should be true & this should be opt-out.
All mentions of MetricsLogger have also been removed.

I've also updated the Kafka broker for CDK container deployment and docker-compose to 3.8 and now use the new Apache Kafka image (which has recently come into being).  MSK still uses 3.6, which is the latest that MSK supports at present.  We no longer rely upon zookeeper for Kafka containers.  Instead, KRaft is enabled, making the deployments a bit simpler and giving us one less thing to debug.  Some other Kafka related stuff (like volumes) were also cleaned up to simplify the developer experience.  The aws otel collector was also upgraded to 0.38.0.

* Category Maintenance, as Otel metrics and Traces via contexts replace the old MetricsLogger stuff.
* Why these changes are required? The OpenSearch analytics domain and the metrics that it was tracking are obviated by the cloudwatch metrics that the otel-collector is now sending out.
* What is the old behavior before changes and new behavior after changes? CDK will no longer deploy an extra domain and bastion.

### Issues Resolved
https://opensearch.atlassian.net/browse/MIGRATIONS-1475

### Testing
Deployed to my AWS account with this cdk.context.json
```
{
  "demo-deploy": {
    "stage": "dev",
    "openAccessPolicyEnabled": true,
    "domainRemovalPolicy": "DESTROY",
    "enableDemoAdmin": true,
    "trafficReplayerEnableClusterFGACAuth": true,
    "captureProxyESServiceEnabled": true,
    "fetchMigrationEnabled": true,
    "otelCollectorEnabled": true,
    "kafkaBrokerServiceEnabled": true,
    "elasticsearchServiceEnabled": true,
    "sourceClusterEndpoint": "https://capture-proxy-es.migration.dev.local:19200",
    "osContainerServiceEnabled": true
  },
  "availability-zones:account=897272129541:region=us-east-1": [
    "us-east-1a",
    "us-east-1b",
    "us-east-1c",
    "us-east-1d",
    "us-east-1e",
    "us-east-1f"
  ],
  "availability-zones:account=897272129541:region=us-east-2": [
    "us-east-2a",
    "us-east-2b",
    "us-east-2c"
  ]
}
```

I ran `cdk --c contextId=demo-deploy --require-approval never --concurrency 3 deploy \*`

### Check List
- [ ] New functionality includes testing
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by AndreKurait** [comment]
is something weird with the commit history here? All my commits are showing up as unverified

File: TrafficCapture/dockerSolution/src/main/docker/otelCollector/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,4 @@
-FROM public.ecr.aws/a0w2c5q7/otelcol-with-opensearch:amd-latest
-#FROM public.ecr.aws/aws-observability/aws-otel-collector:v0.37.0
+FROM public.ecr.aws/aws-observability/aws-otel-collector:v0.37.0
```
Comment (AndreKurait) [inline TrafficCapture/dockerSolution/src/main/docker/otelCollector/Dockerfile:1]: Thank for this

**Review by AndreKurait — APPROVED** [review]

File: deployment/cdk/opensearch-service-migration/lib/stack-composer.ts [hunk 2]
```diff
@@ -179,22 +179,7 @@ export class StackComposer {
         const sourceClusterEndpoint = this.getContextForType('sourceClusterEndpoint', 'string', defaultValues, contextJSON)
         const osContainerServiceEnabled = this.getContextForType('osContainerServiceEnabled', 'boolean', defaultValues, contextJSON)
 
-        const migrationAnalyticsServiceEnabled = this.getContextForType('migrationAnalyticsServiceEnabled', 'boolean', defaultValues, contextJSON)
```
Comment (lewijacn) [inline deployment/cdk/opensearch-service-migration/lib/stack-composer.ts:182]: We have an additional section in the `options.md`  file with these options, can you remove that specific section as well

File: deployment/cdk/opensearch-service-migration/test/stack-composer-ordering.test.ts [hunk 3]
```diff
@@ -100,13 +99,13 @@ test('Test no migration services get deployed when disabled', () => {
...
         "kafkaBrokerServiceEnabled": false,
         "kafkaZookeeperServiceEnabled": false,
-        "migrationAnalyticsServiceEnabled": false
+        "otelCollectorEnabled": false
     }
 
     const stacks = createStackComposer(contextOptions)
 
     const services = [CaptureProxyESStack, CaptureProxyStack, ElasticsearchStack, MigrationConsoleStack,
-        TrafficReplayerStack, KafkaBrokerStack, KafkaZookeeperStack, MigrationAnalyticsStack]
```
Comment (lewijacn) [inline deployment/cdk/opensearch-service-migration/test/stack-composer-ordering.test.ts:109]: Seems like we should replace with OtelCollectorStack right?

File: test/awsE2ESolutionSetup.sh [hunk 4]
```diff
@@ -15,7 +15,7 @@ prepare_source_nodes_for_capture () {
   # Substitute @ to be used instead of ',' for cases where ',' would disrupt formatting of arguments, i.e. AWS SSM commands
   kafka_brokers=$(echo "$kafka_brokers" | tr ',' '@')
   kafka_sg_id=$(aws ssm get-parameter --name "/migration/$deploy_stage/default/trafficStreamSourceAccessSecurityGroupId" --query 'Parameter.Value' --output text)
-  otel_sg_id=$(aws ssm get-parameter --name "/migration/$deploy_stage/default/analyticsDomainSGId" --query 'Parameter.Value' --output text)
+  otel_sg_id=$(aws ssm get-parameter --name "/migration/$deploy_stage/default/
```
Comment (lewijacn) [inline test/awsE2ESolutionSetup.sh:18]: Looks like this line got cut off from what I can tell

**Review by lewijacn — CHANGES_REQUESTED** [review]
Thanks for cleaning this up! A few additional minor things we should do:
* Remove the analytics options from the `default-values.json`
* Add the otel collector enabled option to the `default-values.json`
* Remove the `accessAnalyticsDashboard.sh` file

**Review by lewijacn — APPROVED** [review]

**Pull request merged**